src/Core/Managed/Shared/Extensibility/MetricManager.cs

```diff
@@ -119,12 +119,6 @@ public void Flush()
         /// </summary>
         public void Dispose()
         {
-            if (this.snapshotTimer != null)
-            {
-                this.snapshotTimer.Dispose();
-                this.snapshotTimer = null;
-            }
-
             this.Flush();
```

Comment:
With the risk of kicking the dead horse, I disagree. Since it's introduction for deterministic release of unmanaged resources, the disposable pattern has evolved to general purpose lifetime management in modern .NET APIs. Any class with special lifetime considerations can implement the `IDisposable` interface to its advantage. Take the `MetricManager` as an example. The removal of `IDisposable` implementation from `TaskTimer` triggered the removal of its disposal from `MetricManager`. Now the `MetricManager.Dispose` should `Cancel` the `snapshotTimer` instead of disposing it; otherwise the `SnapshotAndReschedule` will continue running indefinitely. If `MetricManager` is no longer an `IDisposable`, it'll still need to be "cancellable", "stoppable", etc. because it still needs a special lifetime management.